include/board.h

```diff
@@ -88,12 +91,17 @@ class Board
   virtual void clock_delay(uint32_t milliseconds) = 0;
 
 // serial
-  virtual void serial_init(uint32_t baud_rate, uint32_t dev) = 0;
+  virtual void serial_init(uint32_t baud_rate, hardware_config_t configuration) = 0;
   virtual void serial_write(const uint8_t *src, size_t len) = 0;
   virtual uint16_t serial_bytes_available() = 0;
   virtual uint8_t serial_read() = 0;
   virtual void serial_flush() = 0;
 
+// hardware config
+  virtual bool enable_device(device_t device, hardware_config_t configuration, const Params &params)=0;
+  virtual void init_board_config_manager(ROSflight *rf) = 0;
```

Comment:
I think this will become unnecessary if we remove the `ROSflight` pointer in the config manager per my suggestion above